### PR TITLE
Reorganize animated classes

### PR DESCRIPTION
**Pull request by naftalibeder**
This fixes a bug caused by #2161, where `AnimatedPoint` and `AnimatedShape` were not properly exported.

In working on this, I realized that the `Animated*` objects in `utils` are classes, not utilities - a meaningful difference IMO - and I wanted to differentiate between classes/data and Mapbox markers, which were both contained in `utils`, so I did a bit of a reorganization.

This also deprecates (but still supports) `AnimatedShape` in favor of the more accurate name `AnimatedLineString`.

There were also some internal imports and tests that needed fixing.

This shouldn't break anything except full-path imports, e.g. `import AnimatedPoint from '@rnmapbox/maps/javascript/utils/animated/AnimatedPoint`, which I don't think should be something that we need backwards compatibility for.

### CODE REVIEW

File: javascript/utils/animated/Animated.js [hunk 1]
```diff
@@ -28,12 +23,6 @@ const Animated = {
   SymbolLayer: RNAnimated.createAnimatedComponent(SymbolLayer),
   RasterLayer: RNAnimated.createAnimatedComponent(RasterLayer),
   BackgroundLayer: RNAnimated.createAnimatedComponent(BackgroundLayer),
-
-  // values
-  CoordinatesArray: AnimatedCoordinatesArray,
```
Comment (mfazekas) [inline javascript/utils/animated/Animated.js:33]: Isn't it a breaking change to remove those?!

File: javascript/index.js [hunk 2]
```diff
@@ -124,12 +128,14 @@ export {
...
   Atmosphere,
+  AnimatedCoordinatesArray,
+  AnimatedExtractCoordinateFromArray,
+  AnimatedPoint,
+  AnimatedRouteCoordinatesArray,
+  AnimatedShape,
   locationManager,
   offlineManager,
   snapshotManager,
-  AnimatedMapPoint,
```
Comment (mfazekas) [inline javascript/index.js:130]: AnimatedMapPoint => AnimatedPoint is a breaking change

File: plugin/build/withMapbox.js [hunk 3]
```diff
@@ -48,6 +48,9 @@ function applyCocoaPodsModifications(contents, { RNMapboxMapsImpl, RNMapboxMapsD
     src = addInstallerBlock(src, 'post');
     src = addMapboxInstallerBlock(src, 'pre');
     src = addMapboxInstallerBlock(src, 'post');
+    if (src === '') {
+        return 'no title';
+    }
```
Comment (mfazekas) [inline plugin/build/withMapbox.js:53]: Not sure about that

Comment (naftalibeder) [reply plugin/build/withMapbox.js:53]: Oops that was a mistake, thought I reverted it 😳

File: javascript/utils/animated/Animated.js (hunk 1 above)
Comment (naftalibeder) [reply javascript/utils/animated/Animated.js:33]: Ah true. The animated stuff in particular has so many different import paths. If you want to keep support for all the different imports, would you be all right if I add them back in with a deprecated tag?

I know as a user of `AnimatedPoint` etc., I’ve often found it confusing to not just have a simple import like most elements in this library, e.g. `import { AnimatedPoint } from ‘@rnmapbox/maps’`

Comment (mfazekas) [reply javascript/utils/animated/Animated.js:33]: Yes if we can deprecate them that's fine. We can also add a `deprecatedClass` to `utils/deprecation.js` so we could do something like:

```js
   CoordinatesArray: deprecatedClass(AnimatedCoordinatesArray, "Animated.CoordinatesArray is deprecated, please use AnimatedCoordinatesArray"),
```

```js
function deprecatedClass(klass,message) {
  return class extends klass {
    constructor(...args) {
      console.log(`Deprecated: ${message}`)
      super(...args)
    }
  }
}
```

File: plugin/build/withMapbox.js (hunk 3 above)
Comment (naftalibeder) [reply plugin/build/withMapbox.js:53]: Oh yeah weird - this is generated code.

**Comment by naftalibeder** [comment]
@mfazekas I think this should be good to go.

**Review by mfazekas — CHANGES_REQUESTED** [review]
Sorry I forget to post my comment, pls rename back to AnimatedShape, otherwise looks great!

File: javascript/classes/AnimatedLineString.js [hunk 4]
```diff
@@ -6,23 +6,23 @@ const AnimatedWithChildren = Object.getPrototypeOf(Animated.ValueXY);
 if (__DEV__) {
   if (AnimatedWithChildren.name !== 'AnimatedWithChildren') {
     console.error(
-      'AnimatedShape could not obtain AnimatedWithChildren base class',
```
[the remark this answers is not in the document]
Comment (naftalibeder) [reply javascript/classes/AnimatedLineString.js]: Sorry for the wait - this should be good to go, lmk!

**Review by mfazekas — APPROVED** [review]

**Review by mfazekas — APPROVED** [review]
LGTM

There are some lint issues to be fixed

**Comment by naftalibeder** [comment]
All right, everything works locally for me. For some reason the test action just failed with the error relating to `uncommitted changes`, which is weird for the CI. Running again...

I updated a few dependencies related to testing, can you verify you're okay with everything?

File: __tests__/components/UserLocation.test.js [hunk 5]
```diff
@@ -34,33 +34,29 @@ describe('UserLocation', () => {
...
+      const circleLayer = UNSAFE_getAllByType(CircleLayer);
 
-        expect(shapeSource.length).toBe(1);
-        expect(circleLayer.length).toBe(3);
-        done();
-      });
-    });
+      expect(shapeSource.length).toBe(1);
+      expect(circleLayer.length).toBe(3);
+    }, 1000);
```
Comment (mfazekas) [inline __tests__/components/UserLocation.test.js:45]: 👍  awesome! thanks for correcting the async tests

```suggestion
    });
```

Do we need test specific timeout?! We could also lower the default timeout can't we?!

**Review by mfazekas — APPROVED** [review]
Looks good to me, the CI still needs to be fixed.

File: .eslintrc.js [hunk 6]
```diff
@@ -38,8 +38,10 @@ module.exports = {
   extends: [
     'eslint:recommended',
     'plugin:react/recommended',
+    'plugin:jest/recommended',
     '@react-native-community',
     'prettier',
+    'plugin:jest/recommended',
```
[the remark this answers is not in the document]
Comment (naftalibeder) [reply .eslintrc.js]: Whoops! 😆

File: __tests__/components/UserLocation.test.js (hunk 5 above)
Comment (naftalibeder) [reply __tests__/components/UserLocation.test.js]: That's right, that was left over from a different fix attempt. Good eye!

**Comment by naftalibeder** [comment]
OK, merging, thanks!

**Comment by mfazekas** [comment]
@naftalibeder pls fix CI too

**Pull request merged**